_rules/table-headers-attribute-refer-to-data-cells-a25f45.md

```diff
@@ -0,0 +1,295 @@
+			<td headers="headerAge">65</td>
+		</tr>
+	</tbody>
+</table>
+```
+
+#### Passed Example 6
+
+The `headers` attribute on the last two `th` elements refers to another `th` element within the same `table`. Here the column header has a span of two columns.
+
+```html
+<table>
+	<tr>
+		<th id="name" colspan="2">Name</th>
+	</tr>
+	<tr>
+		<th headers="name">Firsname</th>
+		<th headers="name">Lastname</th>
+	</tr>
+</table>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The `headers` attribute on the cell refers to an element outside the same `table`.
+
+```html
+<span id="elmOutsideTable">Project Costs</span>
+
+<table>
+	<tr>
+		<th id="headerOfColumn">Projects</th>
+	</tr>
+	<tr>
+		<td headers="elmOutsideTable">15%</td>
+	</tr>
+</table>
+```
+
+#### Failed Example 2
+
+The `td` element has a `headers` attribute referring to an ID that does not exist within the same `table` (or within the [document](https://dom.spec.whatwg.org/#concept-document)). Even if some of the ID referred to exist, a single non-existing one is enough to fail the rule.
+
+```html
+<table>
+	<tr>
+		<th colspan="2" id="header1">Projects</th>
+	</tr>
+	<tr>
+		<th id="e1" headers="header1">1</th>
+		<th id="e2" headers="header1">2</th>
+	</tr>
+	<tr>
+		<td colspan="2" headers="header1 e1 NOT-EXIST">15%</td>
+	</tr>
+</table>
+```
+
+#### Failed Example 3
+
+The `headers` attribute on the cell refers to an element inside the same `table` which does not have a role of `rowheader` or `columnheader`.
+
+```html
+<table>
+	<tr>
+		<td>
+			<span id="headerProject">Projects</span>
+		</td>
+	</tr>
+	<tr>
+		<td headers="headerProject">
+			15%
+		</td>
+	</tr>
+</table>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+There is no `headers` attribute.
+
+```html
+<table>
+	<tr>
+		<th scope="col">Projects</th>
+		<th scope="col">Exams</th>
+	</tr>
+	<tr>
+		<td>15%</td>
+		<td>45%</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 2
+
+The `table` has a `role="presentation"` and thus is not [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+```html
+<table role="presentation">
+	<tr>
+		<td id="header1">Project Status</td>
+	</tr>
+	<tr>
+		<td>15%</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 3
+
+The `table` is not [visible](#visible) in page.
+
+```html
+<html>
+	<style>
+		.notInPage {
+			position: absolute;
+			left: -9999px;
+			top: -9999px;
+		}
+	</style>
+	<table class="notInPage">
+		<tr>
+			<td id="header1">Project Status</td>
+		</tr>
+		<tr>
+			<td>15%</td>
+		</tr>
+	</table>
+</html>
+```
+
+#### Inapplicable Example 4
+
+The rule applies only to `headers` attribute within a `table` element.
+
+```html
+<div role="table">
+	<div role="row">
+		<div role="columnheader" id="header1">Projects</div>
+		<div role="columnheader" id="header2">Exams</div>
+	</div>
+	<div role="row">
+		<div role="cell" headers="header2">15%</div>
+		<div role="cell" headers="header1">15%</div>
+	</div>
+</div>
+```
```

Comment:
Will a different rule check this kind of thing? This code could be styled to look just like a table, and has all the correct attributes.

Reply:
Yes, it will be tested by a different rule. The headers attribute does not work on div. It can only be on td / th. This should be in the description.